### ISSUE TITLE
Question about compiler warnings

### ISSUE THREAD

**Issue by bjadamson** [libsdl-org/SDL#10080]
These warnings have been popping up in trunk lately, is it desired that these are caught by CI? I know it's sometimes annoying to fail a build on unused variables, but I wanted to find out the policy.

![image](https://github.com/libsdl-org/SDL/assets/1087837/b99b51d0-3107-4dcd-b13a-6592b29dff44)

**Comment by madebr** [comment]
These warnings appeared because you configured SDL without ibus or fcitx support.
They did not occur on ci because those development packages are always installed.
We do not have a way to test every permutation.

This will fix your warning, but I don't know whether this is the correct fix:
```patch
--- a/src/video/x11/SDL_x11events.c
+++ b/src/video/x11/SDL_x11events.c
@@ -887,8 +887,10 @@ void X11_HandleKeyEvent(SDL_VideoDevice *_this, SDL_WindowData *windowdata, SDL_
         /* Save the original keycode for dead keys, which are filtered out by
            the XFilterEvent() call below.
         */
+#if defined(HAVE_IBUS_IBUS_H) || defined(HAVE_FCITX)
         int orig_event_type = xevent->type;
         KeyCode orig_keycode = xevent->xkey.keycode;
+#endif
 
         /* filter events catches XIM events and sends them to the correct handler */
         if (X11_XFilterEvent(xevent, None)) {
```

@slouken touched this in 658f3cdcf1a480b055fe81847fcbdaa557dc7a8b

**Comment by bjadamson** [comment]
I see -- so should I just report these issues in the future like this then?

**Comment by madebr** [comment]
Yes please!
Being correct and warning-free remaing a goal.
I war merely explaining why the warning did not appear on ci.

**Comment by slouken** [comment]
> These warnings appeared because you configured SDL without ibus or fcitx support. They did not occur on ci because those development packages are always installed. We do not have a way to test every permutation.
> 
> This will fix your warning, but I don't know whether this is the correct fix:
> 
> ```diff
> --- a/src/video/x11/SDL_x11events.c
> +++ b/src/video/x11/SDL_x11events.c
> --- a/src/video/x11/SDL_x11events.c
> +++ b/src/video/x11/SDL_x11events.c
> @@ -884,12 +884,14 @@ void X11_HandleKeyEvent(SDL_VideoDevice *_this, SDL_WindowData *windowdata, SDL_
>      text[0] = '\0';
>  
>      if (SDL_TextInputActive()) {
> +#if defined(HAVE_IBUS_IBUS_H) || defined(HAVE_FCITX)
>          /* Save the original keycode for dead keys, which are filtered out by
>             the XFilterEvent() call below.
>          */
>          int orig_event_type = xevent->type;
>          KeyCode orig_keycode = xevent->xkey.keycode;
>  
> +
>          /* filter events catches XIM events and sends them to the correct handler */
>          if (X11_XFilterEvent(xevent, None)) {
>  #if 0
> ```
> 
> @slouken touched this in [658f3cd](https://github.com/libsdl-org/SDL/commit/658f3cdcf1a480b055fe81847fcbdaa557dc7a8b)

This is the correct fix, please feel free to commit. Thanks!

**Comment by bjadamson** [comment]
> Yes please! Being correct and warning-free remaing a goal. I war merely explaining why the warning did not appear on ci.

Ok! thanks (:

**Comment by madebr** [comment]
This is fixed in cf267e498

Thanks!
